agent/exec/container/container.go

```diff
@@ -83,9 +83,11 @@ func (c *containerConfig) image() string {
 }
 
 func (c *containerConfig) config() *enginecontainer.Config {
+	var hostname = strings.Replace(c.spec().Hostname, "*", fmt.Sprint(c.task.Slot), 1)
```

Comment:
We need to be careful about creating a "template" system without thinking about consistent user experience.

See https://github.com/docker/docker/issues/24110#issuecomment-234358152 for wider discussion on this.

I know this is a desired feature but there are a lot implications to whatever choice we make.